src/net_ah_fmt_plug.c

```diff
@@ -0,0 +1,264 @@
+/* Cracker for IPsec Authentication Header (AH) hashes.
+ *
+ * This software is Copyright (c) 2017, Dhiru Kholia <dhiru [at] openwall.com>,
+ * and it is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_netah;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_netah);
+#else
+
+#include <string.h>
+#ifdef _OPENMP
+#include <omp.h>
+#ifndef OMP_SCALE
+#define OMP_SCALE               2048
+#endif
+#endif
+
+#include "formats.h"
+#include "hmacmd5.h"
+#include "misc.h"
+#include "common.h"
+#include "params.h"
+#include "options.h"
+#include "memdbg.h"
+
+#define FORMAT_LABEL            "net-ah"
+#define FORMAT_NAME             "IPsec AH HMAC-MD5-96"
+#define FORMAT_TAG              "$net-ah$"
+#define TAG_LENGTH              (sizeof(FORMAT_TAG) - 1)
+#define ALGORITHM_NAME          "MD5 32/" ARCH_BITS_STR
+#define BENCHMARK_COMMENT       ""
+#define BENCHMARK_LENGTH        0
+#define PLAINTEXT_LENGTH        16  // actual password length limit seems to be 8
```

Comment:
Try using a length 8 password ending in eg. a `£`. In UTF-8, that character is 0xc2, 0xa3 so the password is 8 characters but 9 bytes. I bet it will be truncated after 0xc2.